### ISSUE TITLE
Open URL: doesn't work

### ISSUE THREAD

**Issue by p-mackay** [mpv-android/mpv-android#840]
### Important information

Android version: 14

mpv-android version: 2023-11-30-release

### Description
1 - I tried to use the "OPEN URL" feature on both the builds from f-droid, and the aurora store. In both cases when I try to play a url video nothing happens. It just closes out (back to the mpv home screen).

2 - Then since those both weren't working I tried uninstalling and installing the following APK: https://github.com/mpv-android/mpv-android/pull/58. 
- I try to play a URL video with this build and I get the warning: "To play content from video platforms youtube-dl needs to be set up first..."
- Go into settings > Advanced > Install/Update youtube-dl > INSTALL. "Installing yt-dlp: /data/user/0/is.xyz/files/ytdl/wrapper[5]: ./python3: not executable: 32-bit ELF file"
- App info > Open by default > "all youtube options"... still doesn't work

Am I missing something obvious? Thank you for your time.

**Comment by sfan5** [comment]
You might not be missing anything obvious but I am. You didn't include a log file or any precise description of what kind of "video URL" you are trying to play.

The kind of URLs mpv can play out of the box are *direct* links to `mp4`, `mkv`, `m3u` files or similar.
When you enter `https://www.youtube.com/watch?v=dQw4w9WgXcQ` that's just a HTML page. mpv can't play a HTML page.
youtube-dl's (or today yt-dlp) job is to extract the actual video stream URL from there and give it to mpv, which is why you need it.

Now for why it doesn't work:
> "Installing yt-dlp: /data/user/0/is.xyz/files/ytdl/wrapper[5]: ./python3: not executable: 32-bit ELF file"

probably either
* google added additional security restrictions even for "old" apps that prevent this from working
* you are on a device that can't run 32-bit ARM executables

**Comment by p-mackay** [comment]
### Log file:
[mpv.log](https://github.com/mpv-android/mpv-android/files/13802511/mpv.log)
### Additional information:
- Google Pixel 8
- Graphine OS (shiba)
- URL type: https://www.youtube.com/watch?v=dQw4w9WgXcQ

Given the above information is there any way to get the "OPEN URL" feature to work on my device?

**Comment by sfan5** [comment]
> Graphine OS (shiba)

You mean Graphene OS?
Being a security-focused Android variant I imagine they plugged this "hole" ages ago.

tl;dr no

**Comment by algmyr** [comment]
I think this is actually pixel 8's new tensor chip dropping 32-bit support. I see the same issue on a stock pixel 8 pro. Would it be feasible to build a 64 bit python and create an adjusted apk to test this?

**Comment by sfan5** [comment]
> Would it be feasible to build a 64 bit python and create an adjusted apk to test this?

Definitely. Shouldn't be hard either.
I don't plan to spend my time on this however, you can find the scripts in [this branch](https://github.com/mpv-android/mpv-android/tree/sfan5/python).

**Comment by syphyr** [comment]
I've set the default to arm64 and everything works with python and openssl.  
https://github.com/syphyr/mpv-android/commit/89fd90f53fdb25105a491826597dc9a5f56bd539

With this commit, `buildall.sh` will default to building a complete arm64 apk.

**Comment by kpg-anon** [comment]
> I've set the default to arm64 and everything works with python and openssl. [syphyr@2e752b2](https://github.com/syphyr/mpv-android/commit/2e752b221d631a63cb37d256de79cddafc478ff3)
> 
> With this commit, `buildall.sh` will default to building a complete arm64 apk.

Can you provide an arm64 apk? I tried to build it myself using the instructions [here](https://github.com/syphyr/mpv-android/blob/master/buildscripts/README.md) but ran into all kinds of errors.

**Comment by syphyr** [comment]
> > I've set the default to arm64 and everything works with python and openssl. [syphyr@2e752b2](https://github.com/syphyr/mpv-android/commit/2e752b221d631a63cb37d256de79cddafc478ff3)
> > With this commit, `buildall.sh` will default to building a complete arm64 apk.
> 
> @syphyr Can you provide an arm64 apk? I tried to build it myself using the instructions [here](https://github.com/syphyr/mpv-android/blob/master/buildscripts/README.md) but ran into all kinds of errors.

[mpv-android-v0.38.0-486-g3fa09458da-arm64-v8a-release.apk](https://mega.nz/file/zNR3GZaL#kF0NNItLYbQtnfliUNhgkzBmOKtMyOVfNDNpWZyc534)

**Comment by kpg-anon** [comment]
> > > I've set the default to arm64 and everything works with python and openssl. [syphyr@2e752b2](https://github.com/syphyr/mpv-android/commit/2e752b221d631a63cb37d256de79cddafc478ff3)
> > > With this commit, `buildall.sh` will default to building a complete arm64 apk.
> > 
> > 
> > @syphyr Can you provide an arm64 apk? I tried to build it myself using the instructions [here](https://github.com/syphyr/mpv-android/blob/master/buildscripts/README.md) but ran into all kinds of errors.
> 
> [mpv-android-v0.38.0-486-g3fa09458da-arm64-v8a-release.apk](https://mega.nz/file/zNR3GZaL#kF0NNItLYbQtnfliUNhgkzBmOKtMyOVfNDNpWZyc534)

Confirmed working on Android 14. Thank you

**Comment by sebma** [comment]
@sfan5 Hi, I can see you closed this issue as completed on the [1st of January 2024](https://github.com/mpv-android/mpv-android/issues/840#event-11365793017), what is the PR / commit that has solved this issue ?

I see no `yt-dlp` installation menu in `mpv-android` 2024-06-05 buid ?

What release of `mpv-android` contains the `yt-dlp` installation menu ?

**Comment by sfan5** [comment]
**Only** the build from #58 contains support for yt-dlp. This hasn't changed in years.
